app/controllers/foreman_deployments/api/v2/deployments_controller.rb

```diff
@@ -0,0 +1,59 @@
+module ForemanDeployments
+  module Api
+    module V2
+      class DeploymentsController < BaseController
+        before_filter :find_resource, :only => [:show, :configure, :configuration, :run]
+
+        rescue_from ForemanDeployments::Config::InvalidValueException, :with => :unprocessable_entity_error
+
+        def_param_group :deployment do
+          param :deployment, Hash, :required => true, :action_aware => true do
+            param :name, String, :required => true, :desc => N_('Name for the deployment')
+            param :stack_id, :identifier, :required => true, :desc => N_('Id of the stack to deploy')
+          end
+        end
+
+        api :POST, '/deployments/', N_('Create a deployment')
+        param_group :deployment, :as => :create
+        def create
+          deployment_params = params[:deployment]
+
+          stack = ForemanDeployments::Stack.find(deployment_params.delete(:stack_id)) if deployment_params[:stack_id]
+          deployment_params[:configuration] = ForemanDeployments::Configuration.new(:stack => stack)
+
+          @deployment = ForemanDeployments::Deployment.new(deployment_params)
+          process_response @deployment.save
+        end
+
+        api :GET, '/deployments/', N_('List deployments')
+        def index
+          @deployments = resource_scope
+        end
+
+        api :PUT, '/deployments/:id/configuration/', N_('Configure a deployment')
```

Comment:
REST dictates that we use PUT for a full update (replace) semantics, and POST as a factory/changeset semantics. Can we differentiate PUT to call `#configure` and POST to call `#merge_configuration`?